CommunicationLayer/CommDeviceControl/InternetConnectionService.cpp

```diff
@@ -0,0 +1,82 @@
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+#include <QUdpSocket>
+#include <QDebug>
+#include <QTextStream>
+#include <QTimer>
+
+#include "../../InfrastructureLayer/Settings/I_Settings.h"
+#include "InternetConnectionService.h"
+namespace
+{
+    quint32 SLEEP_TIME_MILLISECONDS = 2000;
+}
+
+InternetConnectionService::InternetConnectionService(
+    QString exchangeName
+    ,  QString ipAddress
+    ,  quint16 port)
+    : exchangeName_(exchangeName)
+    , ipAddress_(ipAddress)
+    , port_(port)
+    , connectionRetryTimer_(new QTimer(this))
+{
+    QObject::connect(this, SIGNAL(setupChannelSignal()), this, SLOT(setupChannel()));
+
+    //connectionRetryTimer_& = new QTimer(this);
+    connectionRetryTimer_.setSingleShot(true);
+    connect(&connectionRetryTimer_, SIGNAL(timeout()), this, SLOT(setupChannel()));
```

Comment:
I don't think this will attempt connection right away, but waits for one delay period. I don't think that's optimal behaviour.